plugin/storage/kafka/writer_test.go

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) 2018 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kafka
+
+import (
+	"testing"
+	"time"
+
+	"github.com/Shopify/sarama"
+	saramaMocks "github.com/Shopify/sarama/mocks"
+	"github.com/davit-y/jaeger/storage/spanstore"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/uber/jaeger-lib/metrics"
+	"github.com/uber/jaeger-lib/metrics/testutils"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/pkg/kafka/mocks"
+)
+
+type spanWriterTest struct {
+	producer       *saramaMocks.AsyncProducer
+	marshaller     *mocks.Marshaller
+	metricsFactory *metrics.LocalFactory
+
+	writer *SpanWriter
+}
+
+// Checks that Kafka SpanWriter conforms to spanstore.Writer API
+var _ spanstore.Writer = &SpanWriter{}
+
+func withSpanWriter(t *testing.T, fn func(span *model.Span, w *spanWriterTest)) {
+	serviceMetrics := metrics.NewLocalFactory(100 * time.Millisecond)
+	saramaConfig := sarama.NewConfig()
+	saramaConfig.Producer.Return.Successes = true
+	producer := saramaMocks.NewAsyncProducer(t, saramaConfig)
+	marshaller := &mocks.Marshaller{}
+	marshaller.On("Marshal", mock.AnythingOfType("*model.Span")).Return([]byte{}, nil)
+
+	tags := model.KeyValues{
+		model.String("someStringTagKey", "someStringTagValue"),
+	}
+	writerTest := &spanWriterTest{
+		producer:       producer,
+		marshaller:     marshaller,
+		metricsFactory: serviceMetrics,
+		writer:         NewSpanWriter(producer, marshaller, "someTopic", serviceMetrics),
+	}
+
+	span := &model.Span{
```

Comment:
is there anything special about this span? Can we move it out into a var at the top of the file and share it with the marshaller test?